lib/internal/test_runner/reporter/tap.js

```diff
@@ -143,12 +143,12 @@ function tapEscape(input) {
   return result;
 }
 
-function jsToYaml(indent, name, value) {
+function jsToYaml(indent, name, value, stringify = typeof value !== 'object') {
```

Comment:
I think `jsToYaml()` needs to take a context object (which `indent` and this new argument could be part of). The context can track which fields have been seen before to prevent duplicates.

Reply:
not sure how that would solve this issue,
if we take the example in https://github.com/nodejs/node/issues/47075
why would you expect the YAML to have a `foo` key?

Reply:
I wouldn't expect it to have a `foo` key by default, but [this loop](https://github.com/nodejs/node/blob/334bb179dfef5ad51e3dc1a7925a00b07bedada6/lib/internal/test_runner/reporter/tap.js#L177-L185) prints all of the fields in the error or object. If the object had, for example, a `duration_ms` property, then wouldn't we end up with duplicates again because of [this logic](https://github.com/nodejs/node/blob/334bb179dfef5ad51e3dc1a7925a00b07bedada6/lib/internal/test_runner/reporter/tap.js#L115-L116)? It looks like it to me based on https://github.com/nodejs/node/issues/47075.